src/Particle/VirtualParticleSet.cpp

```diff
@@ -112,7 +113,35 @@ void VirtualParticleSet::makeMoves(int jel,
   refSourcePtcl = iat;
   assert(R.size() == deltaV.size());
   for (size_t ivp = 0; ivp < R.size(); ivp++)
-    R[ivp] = ref_pos + deltaV[ivp];
+  {
+    R[ivp]     = ref_pos + deltaV[ivp];
+    spins[ivp] = ref_spin; //no spin deltas in this API
```

Comment:
I'm not getting why `spins` needs to be updated. When not using spinor, Should the whole run not depending on the value of spins? Will there be any issue if we keep makeMoves untouched?

Reply:
Not for normal calculations. But for calculations that do include spinors yes it does matter. We still have NonLocalECPs used in spinor calculations, so if those evaluateRatios don't use VPs that have up to date spin values, the ratio evaluations are wrong

Reply:
Basically, if we don't have spinors then the original behavior of makeMoves would be fine. But if we do have spinors, then I need it to also have up to date spins (but no actual displacement from their original values). 

I wasn't sure of the best way to handle this.

Reply:
For reference, if I don't keep the spins up to date in NonLocalECPComponent, then I get the following

> mymac:build_gcc cmelton$ ctest -R SOREP-vmc                              
Test project /Users/cmelton/Software/qmcpack/build_gcc
      Start 1286: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16
 1/24 Test #1286: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16 ......................   Passed   81.97 sec
      Start 1287: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-kinetic
 2/24 Test #1287: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-kinetic ..............   Passed    0.03 sec
      Start 1288: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-totenergy
 3/24 Test #1288: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-totenergy ............***Failed    0.02 sec
      Start 1289: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-eeenergy
 4/24 Test #1289: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-eeenergy .............   Passed    0.02 sec
      Start 1290: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-samples
 5/24 Test #1290: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-samples ..............   Passed    0.02 sec
      Start 1291: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-potential
 6/24 Test #1291: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-potential ............***Failed    0.02 sec
      Start 1292: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-ionion
 7/24 Test #1292: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-ionion ...............   Passed    0.02 sec
      Start 1293: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-localecp
 8/24 Test #1293: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-localecp .............   Passed    0.02 sec
      Start 1294: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-nonlocalecp
 9/24 Test #1294: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-nonlocalecp ..........***Failed    0.02 sec
      Start 1295: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-soecp
10/24 Test #1295: short-bccMo_2x1x1_SOREP-vmc_sdj-1-16-soecp ................   Passed    0.02 sec

Reply:
> so if those evaluateRatios don't use VPs that have up to date spin values, the ratio evaluations are wrong

evaluateRatios should be fixed to handle VP isSpinor() true and false cases. It should not blindly use spins. VP isSpinor() should carry identical value as the ParticleSet it constructs from.

Reply:
Thanks for the info and I have a better picture now. I will make a follow up PR to consolidate APIs.